core/src/scene/sceneLoader.cpp

```diff
@@ -330,6 +349,236 @@ void SceneLoader::loadTextures(Node textures, Scene& scene) {
+    if (materialNode) {
+        loadMaterial(materialNode, *(style->getMaterial()), scene);
+    }
+
+    Node lightingNode = styleNode["lighting"];
+    if (lightingNode) {
+        std::string lighting = lightingNode.as<std::string>();
+        if (lighting == "fragment") { style->setLightingType(LightingType::fragment); }
+        else if (lighting == "vertex") { style->setLightingType(LightingType::vertex); }
+        else if (lighting == "false") { style->setLightingType(LightingType::none); }
+        else if (lighting == "true") { } // use default lighting
+        else { logMsg("WARNING: unrecognized lighting type \"%s\"\n", lighting.c_str()); }
+    }
+
+	Node textureNode = styleNode["texture"];
+    if (textureNode) {
+    	auto spriteStyle = dynamic_cast<SpriteStyle*>(style);
+        if (spriteStyle) {
+        	std::string textureName = textureNode.as<std::string>();
+            auto atlases = scene.spriteAtlases();
+            auto it = atlases.find(textureName);
+            if (it != atlases.end()) {
+            	spriteStyle->setSpriteAtlas(it->second);
+            } else {
+            	logMsg("WARNING: undefined texture name %s", textureName.c_str());
+            }
+        }
+	}
+
+    Node urlNode = styleNode["url"];
+    if (urlNode) { logMsg("WARNING: loading style from URL not yet implemented\n"); } // TODO
+
+}
+
+Node SceneLoader::propORing(const std::string& propStr, Node& styles, const MIXES& mixes) {
+    Node propNode;
+    for (const auto& mixStyleName : mixes) {
+        auto mixStyleNode = styles[mixStyleName];
+        if ( (propNode = mixStyleNode[propStr]) ) {
+            bool val = propNode.as<bool>();
+            if (val) { return propNode; }
+        }
+    }
+    return propNode;
+}
+
+Node SceneLoader::propOverwrite(const std::string& propStr, const std::string& subPropStr, Node& styles,
+        const MIXES& mixes) {
+    Node node;
+    for (const auto& mixStyleName : mixes) {
+        auto mixStyleNode = styles[mixStyleName];
+        if (auto propNode = mixStyleNode[propStr]) {
+            if (subPropStr.size() == 0) { node = propNode; }
+            else if (auto subPropNode = propNode[subPropStr]) { node = subPropNode; }
+        }
+    }
+    return node;
+}
+
+Node SceneLoader::propMerge(const std::string& propStr, const std::string& subPropStr, Node& styles,
+        const MIXES& mixes) {
+
+    std::unordered_set<std::string> uniqueList;
+    Node node;
+    for (const auto& mixStyleName : mixes) {
+        auto mixStyleNode = styles[mixStyleName];
+        if (auto propNode = mixStyleNode[propStr]) {
+            if (auto subPropNode = propNode[subPropStr]) {
+                if (subPropNode.IsMap()) { // for something like shaders: defines or unique
+                    for (const auto& val : subPropNode) {
+                        node[val.first] = val.second;
+                    }
+                } else if (subPropNode.IsScalar()) { // for something like shaders: extension (scalar)
```

Comment:
This is not always the behavior you want for scalars, it's specific to `extensions`. In most cases, scalars overwrite the previous value. The way I would think of `extensions` is to convert a single value into a list, before the merging happens.

I'm weary of having feature-specific code that otherwise reads as generic.